dev_server/server.js

```diff
@@ -0,0 +1,80 @@
+const { createProxyMiddleware } = require('http-proxy-middleware');
+const checkPort = require('./checkPort');
+const router = require('./routes');
+
+function getBaseUrl() {
+  switch (process.env.ENV) {
+    case 'production':
+      return process.env.BASEURL_PRODUCTION;
+    case 'dev':
+      return process.env.BASEURL_DEVELOPMENT;
+    case 'hack':
+      return process.env.BASEURL_HACK;
+    case 'local':
+    default:
+      return process.env.BASEURL_LOCAL;
+  }
+}
+
+const baseurl = getBaseUrl();
+
+module.exports = function (app) {
+  app.use(
+    '/api',
+    createProxyMiddleware({
+      target: baseurl,
+      changeOrigin: true,
+      pathRewrite: {
+        // drop `/api` from path prefix, since its only purpose is to route local requests here
+        '^/api': '/',
+      },
```

Comment:
simplified path rewrite.

basically, since go server expects `/glry/v1/*`, the proxy would cause a bug by dropping this.

now we'll always have the client attach `/glry/v1`, and this is accounted for in our local server here: https://github.com/gallery-so/gallery/pull/283/files#diff-55ded419532a0b5a6cc541f4dd2a15f16e430acbd203359c699738836e3591e2R79